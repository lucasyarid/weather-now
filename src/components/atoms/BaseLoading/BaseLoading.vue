<template>
  <figure class="BaseLoading">
    <img class="BaseLoading__img" :src="src" alt="WEATHERNOW Logo" />
  </figure>
</template>

<script>
export default {
  name: "BaseLoading",
  props: {
    variation: {
      type: String,
      default: "default"
    }
  },
  computed: {
    src() {
      return require("@/assets/images/loader.svg");
    }
  }
};
</script>

<style lang="scss" scoped>
.BaseLoading {
  margin: 0;
  width: 50%;
  height: fit-content;
  margin: 0 auto;
  &__img {
    width: 50%;
  }
}
</style>
