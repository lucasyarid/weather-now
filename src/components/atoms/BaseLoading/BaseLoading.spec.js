import { shallowMount } from "@vue/test-utils";
import BaseLoading from "./BaseLoading";

describe("BaseLoading", () => {
  const WRAPPER = shallowMount(BaseLoading);

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
