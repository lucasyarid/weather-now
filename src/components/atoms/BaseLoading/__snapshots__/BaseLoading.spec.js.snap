// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLoading Componente n√£o alterado 1`] = `"<figure class=\\"BaseLoading\\"><img src=\\"loader.svg\\" alt=\\"WEATHERNOW Logo\\" class=\\"BaseLoading__img\\"></figure>"`;
